src/tests/test_common/hs_repl_test_common.hpp

```diff
@@ -348,7 +349,7 @@ class HSReplTestHelper {
     std::vector< std::string > args_;
     char** argv_;
 
-    std::vector< std::string > dev_list_;
+    std::vector< std::pair< std::string, homestore::HSDevType > > dev_list_;
```

Comment:
Can you explain why we need to pass in DevType? This is not needed as iomgr detects devType and that is the fundamental assuption between HomeStore and IOMgr.

Reply:
in test , we always create a file  and consider it as a physical device , right?
when IOManager see this file, it will mark it as a HSDevType:Data.  so I want to explictly mark it as a HSDevType:Fast Drive. so that we can simulate the real-world use case we will face

Reply:
if you are referring to the file case only (for real drives, iomgr will detect drive type for HomeStore, HomeStore is not supposed to specify drive type as it has no idea (also should not know about) which drive type passed in).

For file case,  there is api called `iomgr::DriveInterface::emulate_drive_type()` provided by iomgr, that is supposed to work with file type and simulate nvme/hdd drive types. It has been used and tested intensively in 1.3 HomeStore/IoMgr.

Bottom line is HomeStore and consumer of homestore, also in real life, shouldn't have knowledge about what drive type is being passed to HomeStore, and we should stick to this rule as it is also what happens in production env.

Reply:
yea, I absolutely agree with you. we need to face the real-world use case. actually , I also considered about this problem when writing this code. but I did not find the code about the device detection and type change logic. 
pls see my comment here
https://github.com/eBay/HomeStore/pull/433#discussion_r1623146648

 so , I just did all my work based on the current code of master branch and focused on the support of miss drive.
 I also have a plan to add that device detection and type change logic in a seperate PR.

Reply:
@yamingk there is some case that iomanager will mistakenly consider a hdd file as a nvme file. 
for example , in we do some test in  virtual machine, we will create a file in /tmp directory as a physical device. actually it is a hdd file.


however, iomanager will consider it as a fast drive since the mount drive name of /tmp/ is `atomicos-root` and it can not find `atomicos-root` in `/sys/dev/block/`.  `atomicos-root` does not have major and minor dev name. so it will be consider as not rotational, and then it is considered as a file_on_nvme.

Reply:
As I commented above, the dtype is detected by iomgr from `get_drive_type` (inside `is_hdd(...)`) from HomeStore.